### DISCUSSION TITLE
How update Xformers and Pytorch❓

### DISCUSSION THREAD

**Discussion by ZeroCool22** [lllyasviel/stable-diffusion-webui-forge#1447] [Q&A] [answered]
In **AUTO's1111** the process is:

Go to: C:\Users\ZeroCool22\Desktop\webui_forge\webui\venv\Scripts

Activate and then:

pip install --upgrade torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124

pip install --upgrade xformers==0.0.28.dev881

-----------------------------------------------------------------------

I did the same for **Forge**:

Go to: C:\Users\ZeroCool22\Desktop\webui_forge\webui\scripts

And same commands.

💡But when i open the WEB interface at the bottom, it still says the old versions. So what is wrong?

**Comment by dongxiat** [comment]
![image](https://github.com/user-attachments/assets/0e4b8699-26ce-4a12-9689-903b1ce19fd5)

`environments `of webui-Forge is `system `folder...so if u want install or upgrade something manually just:
1. copy `environment.bat` to new file `venv.bat`
2. Edit `venv.bat` add a line` cmd /k` then save and close
3. Open `venv.bat ` then run `python -m pip install --upgrade pip ` for first time
4. reopen So now u can `pip install --upgrade xformers==0.0.28.dev881` or install what u wantto `enviroment `of Forge

**Reply by ZeroCool22** [reply]
> ![image](https://private-user-images.githubusercontent.com/75362901/360993601-0e4b8699-26ce-4a12-9689-903b1ce19fd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzExNTQsIm5iZiI6MTcyNDQzMDg1NCwicGF0aCI6Ii83NTM2MjkwMS8zNjA5OTM2MDEtMGU0Yjg2OTktMjZjZS00YTEyLTk2ODktOTAzYjFjZTE5ZmQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE2MzQxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyZDc4MzJkZWVkYmZhNWMwYWMzNzgwMDdjOTQ3ZmEwMzZjNGRhYzUxMjgyY2FlY2NhMmJmOGJiOThlNTBmYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.njhCIs198cGn0M6TJPZRtuEicw6Y0YT68auVUCJIHDQ)
> 
> `environments `of webui-Forge is `system `folder...so if u want install or upgrade something manually just:
> 
>     1. copy `environment.bat` to new file `venv.bat`
> 
>     2. Edit `venv.bat` add a line` cmd /k` then save and close
> 
>     3. Open `venv.bat ` then run `python -m pip install --upgrade pip ` for first time
> 
>     4. reopen So now u can `pip install --upgrade xformers==0.0.28.dev881` or install what u wantto `enviroment `of Forge

But, i must activate it once i open venv.bat?

**Reply by dongxiat** [reply, marked as answer]
> > ![image](https://private-user-images.githubusercontent.com/75362901/360993601-0e4b8699-26ce-4a12-9689-903b1ce19fd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzExNTQsIm5iZiI6MTcyNDQzMDg1NCwicGF0aCI6Ii83NTM2MjkwMS8zNjA5OTM2MDEtMGU0Yjg2OTktMjZjZS00YTEyLTk2ODktOTAzYjFjZTE5ZmQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE2MzQxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyZDc4MzJkZWVkYmZhNWMwYWMzNzgwMDdjOTQ3ZmEwMzZjNGRhYzUxMjgyY2FlY2NhMmJmOGJiOThlNTBmYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.njhCIs198cGn0M6TJPZRtuEicw6Y0YT68auVUCJIHDQ)
> > `environments `of webui-Forge is `system `folder...so if u want install or upgrade something manually just:
> > ```
> > 1. copy `environment.bat` to new file `venv.bat`
> > 
> > 2. Edit `venv.bat` add a line` cmd /k` then save and close
> > 
> > 3. Open `venv.bat ` then run `python -m pip install --upgrade pip ` for first time
> > 
> > 4. reopen So now u can `pip install --upgrade xformers==0.0.28.dev881` or install what u wantto `enviroment `of Forge
> > ```
> 
> But, i must activate it once i open venv.bat?

next time u want to update something or uninstall what u want.... just run a `venv.bat` and paste `your command`
